services/documentation_service.py

```diff
@@ -95,15 +105,44 @@ def create_document_generation_job(
         background_tasks.add_task(
             self.generate_documentation_background,
             document_ref.id,
-            github_url
+            github_url,
+            model
         )
 
         return document_ref
 
+    def get_documentation(self, doc_id: str) -> Dict[str, Any]:
+        document_snapshot = self.firebase_client.get_document(
+            self.firebase_client.TEST_COLLECTION,
+            doc_id
+        )
+
+        if not document_snapshot:
+            raise HTTPException(status_code=http.client.NOT_FOUND, detail=f"Document {doc_id} not found.")
+
+        # adding id field since Firestore does not do it naturally
+        document_dict = document_snapshot.to_dict()
+        document_dict["id"] = document_snapshot.id
+
+        return document_dict
+
+    def get_documentation_with_content(self, doc_id: str) -> Dict[str, Any]:
+        doc = self.get_documentation(doc_id)
+
+        status = DocsStatusEnum(doc.get("status"))
+        blob_url = doc.get("bucket_url")
+
+        # if it is completed get the content, otherwise return None
+        doc["content"] = None
+        if status == DocsStatusEnum.COMPLETED:
+            doc["content"] = self.firebase_client.get_blob(blob_url).download_as_text()
+
+        return doc
+
```

Comment:
Do these need to be separate functions? if so, is there a way to rename `get_documentation` so that `get_documentation_with_content()` can be `get_documentation()` instead?

Reply:
Good point. As I was making this change, I realized this fits better inside of FirebaseClient. I ended up making the existing `get_document`, `add_document` ... functions into internal ones, and adding `add_documentation`, `get_documentation` functions that are specific to our use-case.